apis/v1alpha1/forwarding_target_types.go

```diff
@@ -91,3 +91,55 @@ type TargetWeight int32
 //
 // +k8s:deepcopy-gen=false
 type ForwardToTargetObjectReference = ServicesDefaultLocalObjectReference
+
+// GenericForwardToTarget identifies a target object within a known namespace.
```

Comment:
Any better naming suggestions are most welcome here!

Question, should we consider having a separate Forwarding target type for each Route type?

Reply:
I think a shared type is fine until they diverge more. I've been trying to think of a better prefix than "Generic" here but haven't come up with anything.